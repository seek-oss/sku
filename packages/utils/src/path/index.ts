export { toPosixPath } from './toPosixPath.js';
