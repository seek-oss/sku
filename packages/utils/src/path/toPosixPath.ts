import path from 'node:path';

/**
 * Replaces all win32 path separators (\) with posix path separators (/)
 */
export const toPosixPath = (inputPath: string) =>
  inputPath.replaceAll(path.win32.sep, path.posix.sep);
