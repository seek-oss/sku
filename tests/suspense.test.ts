import path from 'node:path';
import {
  dirContentsToObject,
  runSkuScriptInDir,
  waitForUrls,
  getAppSnapshot,
} from '@sku-private/test-utils';
import type { ChildProcess } from 'node:child_process';
import { createRequire } from 'node:module';

const require = createRequire(import.meta.url);

const appDir = path.dirname(
  require.resolve('@sku-fixtures/suspense/sku.config.js'),
);

const targetDirectory = `${appDir}/dist`;
const url = `http://localhost:8203`;

describe('suspense', () => {
  describe.each(['vite', 'webpack'])('bundler %s', (bundler) => {
    const args =
      bundler === 'vite'
        ? ['--config', 'sku.config.vite.js', '--experimental-bundler']
        : [];
    describe('build and serve', () => {
      let process: ChildProcess;

      beforeAll(async () => {
        await runSkuScriptInDir('build', appDir, args);
        process = await runSkuScriptInDir('serve', appDir);
        await waitForUrls(url);
      });

      afterAll(async () => {
        await process.kill();
      });

      it('should return home page', async () => {
        const app = await getAppSnapshot(url);
        expect(app).toMatchSnapshot();
      });

      it('should generate the expected files', async () => {
        const files = await dirContentsToObject(targetDirectory);
        expect(files).toMatchSnapshot();
      });
    });
  });
});
