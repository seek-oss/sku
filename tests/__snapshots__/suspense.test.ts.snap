// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`suspense bundler vite build and serve should generate the expected files 1`] = `
{
  "production/au/index.html": SCRIPTS: [
    "/vite-client.js",
  ]
CSS: []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      rel="modulepreload"
      href="scripts[0]"
      crossorigin
    >
  </head>
  <body>
    <div id="app">
      <ul>
        <li>
          todo1
        </li>
        <li>
          todo2
        </li>
        <li>
          todo3
        </li>
      </ul>
    </div>
    <script
      id="__SKU_CLIENT_CONTEXT__"
      type="application/json"
    >
      {"dehydratedState":{"mutations":[],"queries":[{"state":{"data":["todo1","todo2","todo3"],"dataUpdateCount":1,"dataUpdatedAt":4323283200000,"error":null,"errorUpdateCount":0,"errorUpdatedAt":0,"fetchFailureCount":0,"fetchFailureReason":null,"fetchMeta":null,"isInvalidated":false,"status":"success","fetchStatus":"idle"},"queryKey":["todos"],"queryHash":"[\\"todos\\"]"}]},"site":"au"}
    </script>
    <script
      type="module"
      src="scripts[0]"
    >
    </script>
  </body>
</html>,
  "vite-client.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;

exports[`suspense bundler vite build and serve should return home page 1`] = `
SCRIPTS: [
  "/vite-client.js",
]
CSS: []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      rel="modulepreload"
      href="scripts[0]"
      crossorigin
    >
  </head>
  <body>
    <div id="app">
      <ul>
        <li>
          todo1
        </li>
        <li>
          todo2
        </li>
        <li>
          todo3
        </li>
      </ul>
    </div>
    <script
      id="__SKU_CLIENT_CONTEXT__"
      type="application/json"
    >
      {"dehydratedState":{"mutations":[],"queries":[{"state":{"data":["todo1","todo2","todo3"],"dataUpdateCount":1,"dataUpdatedAt":4323283200000,"error":null,"errorUpdateCount":0,"errorUpdatedAt":0,"fetchFailureCount":0,"fetchFailureReason":null,"fetchMeta":null,"isInvalidated":false,"status":"success","fetchStatus":"idle"},"queryKey":["todos"],"queryHash":"[\\"todos\\"]"}]},"site":"au"}
    </script>
    <script
      type="module"
      src="scripts[0]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;

exports[`suspense bundler webpack build and serve should generate the expected files 1`] = `
{
  "2.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "2.js.map": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "main.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "main.js.map": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "production/au/index.html": SCRIPTS: [
    "/static/suspense/runtime.js",
    "/static/suspense/2.js",
    "/static/suspense/main.js",
  ]
CSS: []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[0]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[1]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[2]"
    >
  </head>
  <body>
    <div id="app">
      <ul>
        <li>
          todo1
        </li>
        <li>
          todo2
        </li>
        <li>
          todo3
        </li>
      </ul>
    </div>
    <script
      id="__SKU_CLIENT_CONTEXT__"
      type="application/json"
    >
      {"dehydratedState":{"mutations":[],"queries":[{"state":{"data":["todo1","todo2","todo3"],"dataUpdateCount":1,"dataUpdatedAt":4323283200000,"error":null,"errorUpdateCount":0,"errorUpdatedAt":0,"fetchFailureCount":0,"fetchFailureReason":null,"fetchMeta":null,"isInvalidated":false,"status":"success","fetchStatus":"idle"},"queryKey":["todos"],"queryHash":"[\\"todos\\"]"}]},"site":"au"}
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
    >
      []
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
    >
      {"namedChunks":[]}
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[0]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[1]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[2]"
    >
    </script>
  </body>
</html>,
  "runtime.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "runtime.js.map": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;

exports[`suspense bundler webpack build and serve should return home page 1`] = `
SCRIPTS: [
  "/static/suspense/runtime.js",
  "/static/suspense/2.js",
  "/static/suspense/main.js",
]
CSS: []
SOURCE HTML: <!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8">
    <title>
      hello-world
    </title>
    <meta
      name="viewport"
      content="width=device-width, initial-scale=1"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[0]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[1]"
    >
    <link
      data-chunk="main"
      rel="preload"
      as="script"
      href="scripts[2]"
    >
  </head>
  <body>
    <div id="app">
      <ul>
        <li>
          todo1
        </li>
        <li>
          todo2
        </li>
        <li>
          todo3
        </li>
      </ul>
    </div>
    <script
      id="__SKU_CLIENT_CONTEXT__"
      type="application/json"
    >
      {"dehydratedState":{"mutations":[],"queries":[{"state":{"data":["todo1","todo2","todo3"],"dataUpdateCount":1,"dataUpdatedAt":4323283200000,"error":null,"errorUpdateCount":0,"errorUpdatedAt":0,"fetchFailureCount":0,"fetchFailureReason":null,"fetchMeta":null,"isInvalidated":false,"status":"success","fetchStatus":"idle"},"queryKey":["todos"],"queryHash":"[\\"todos\\"]"}]},"site":"au"}
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS__"
      type="application/json"
    >
      []
    </script>
    <script
      id="__LOADABLE_REQUIRED_CHUNKS___ext"
      type="application/json"
    >
      {"namedChunks":[]}
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[0]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[1]"
    >
    </script>
    <script
      async
      data-chunk="main"
      src="scripts[2]"
    >
    </script>
  </body>
</html>
POST HYDRATE DIFFS: NO DIFF
`;
